setup/resources/create_resources.go

```diff
@@ -0,0 +1,57 @@
+package resources
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	applycl "github.com/codeready-toolchain/toolchain-common/pkg/client"
+	"github.com/codeready-toolchain/toolchain-common/pkg/template"
+
+	templatev1 "github.com/openshift/api/template/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var tmpl *templatev1.Template
+
+func CreateFromTemplate(cl client.Client, s *runtime.Scheme, templatePath, username string) error {
+	// get the template from the file
+	if tmpl == nil {
+		var err error
+		if tmpl, err = getTemplateFromFile(s, templatePath); err != nil {
+			return fmt.Errorf("invalid template file: '%s'", templatePath)
+		}
+	}
+
+	userNS := fmt.Sprintf("%s-stage", username)
+	// waiting for each namespace here prevents some edge cases where the setup job can progress beyond the usersignup job and fail with a timeout
+	if err := WaitForNamespace(cl, userNS); err != nil {
+		return err
+	}
+	processor := template.NewProcessor(s)
+	objs, err := processor.Process(tmpl.DeepCopy(), map[string]string{
+		"NAMESPACE": userNS,
+	})
+	if err != nil {
+		return err
+	}
+	applycl := applycl.NewApplyClient(cl, s)
+	for _, obj := range objs {
+		if _, err := applycl.ApplyObject(obj.GetRuntimeObject()); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func getTemplateFromFile(s *runtime.Scheme, filename string) (*templatev1.Template, error) {
```

Comment:
I passed in the old user-workloads.yaml file https://github.com/codeready-toolchain/toolchain-e2e/blob/master/setup/sample/user-workloads.yaml and surprisingly it did not fail with an error. No resources were created in the workspace which is expected since there is no OpenShift template but I believe it should result in an error. I suppose it's failing somewhere in this function

Reply:
ok let me check that

Reply:
fixed in 86b510c: an error is returned if the decoding of the content YAML file did not return an OpenShift template